### PR TITLE
Introducing abstractions: EnvironmentEncryptor and TextEncryptorLocator

### PR DESCRIPTION
**Pull request by rzukow**

### CODE REVIEW

File: spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/encryption/CipherEnvironmentEncryptorTests.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    EnvironmentEncryptor encryptor = new CipherEnvironmentEncryptor(textEncryptorLocator);
+
+
+    @Test
+    public void shouldDecryptEnvironment() {
+        // given
+        textEncryptorLocator.setEncryptor(textEncryptor);
+
+        Environment environment = new Environment("name", "profile");
```
Comment (wojtkiewicz) [inline spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/encryption/CipherEnvironmentEncryptorTests.java]: This variable is not needed, we can use some simple String value as a secret

Comment (wojtkiewicz) [inline spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/encryption/CipherEnvironmentEncryptorTests.java]: test line note

Comment (rzukow) [inline spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/encryption/CipherEnvironmentEncryptorTests.java]: answer

Comment (wojtkiewicz) [inline spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/encryption/CipherEnvironmentEncryptorTests.java]: fixed

**Pull request merged**